sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data/ExponentialHistogramBuckets.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.data;
+
+import java.util.List;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * ExponentialHistogramBuckets represents either the positive or negative measurements taken for a
+ * {@link ExponentialHistogramPointData}.
+ *
+ * <p>The bucket boundaries are lower-bound inclusive, and are calculated using the {@link
+ * ExponentialHistogramPointData#getScale()} and the {@link #getOffset()}.
+ *
+ * <p>For example, assume {@link ExponentialHistogramPointData#getScale()} is 0, implying {@link
+ * ExponentialHistogramPointData#getBase()} is 2.0. Then, if <code>offset</code> is 0, the bucket
+ * lower bounds would be 1.0, 2.0, 4.0, 8.0, etc. If <code>offset</code> is -3, the bucket lower
+ * bounds would be 0.125, 0.25, 0.5, 1.0, 2,0, etc. If <code>offset</code> is +3, the bucket lower
+ * bounds would be 8.0, 16.0, 32.0, etc.
+ */
+@Immutable
+public interface ExponentialHistogramBuckets {
+
+  /**
+   * The offset shifts the bucket boundaries according to <code>lower_bound = base^(offset+i).
+   * </code>.
+   *
+   * @return the offset.
+   */
+  int getOffset();
+
+  /**
+   * The bucket counts is a of counts representing number of measurements that fall into each
+   * bucket.
+   *
+   * @return the bucket counts.
+   */
+  List<Long> getBucketCounts();
```

Comment:
Do we think we need `Long` level numbers for the bucket counts? That's a *lot* of recordings for a single bucket.

Reply:
In the NrSketch implementation there is a [backing array that uses variable-width counts](https://github.com/newrelic-experimental/newrelic-sketch-java/blob/main/src/main/java/com/newrelic/nrsketch/WindowedCounterArray.java).

See also OTel-Go: https://github.com/open-telemetry/opentelemetry-go/pull/2261

Reply:
Looking at the details here, I think exposing `List<Long>` may be... overkill for Java.    (@anuraaga will prove me wrong with some fun optimisation around `List`), but from every example @jmacd lists, I think we want to have the flexibilty to use a primitive array and keep the primitives.    That implies exposing a `long getBucketCountAt(int idx)` vs. an actual list (which is what the Go SDK + NR impl are doing).

Reply:
It would make sense to use something similar to the `WindowedCounterArray` backed by the `MultiTypeCounterArray` from the NrSketch that @jmacd mentioned.

I can replace `List<Long> getBucketCounts()` with `long getBucketCountAt(int idx)`. Though, it does raise the question in my mind of how the entire collection of counts would be gathered easily. I can't see that in the Go SDK. Am I incorrectly assuming that it's a necessary method to have for aggregation/exporting?

Reply:
By the way, I expect marshaling to look like

```java
int size = 0;
for (int i = 0; i < getNumBuckets(); i++) {
  size += CodedOutputStream.computeUint32SizeNoTag(getBucketCountAt(i));
}
writeTag(fieldNumber, size)
for (int i = 0; i < getNumBuckets(); i++) {
  writeUint32NoTag(getBucketCountAt(i);
}
```

Reply:
By the way, just curious if any consideration was given to a sparse array given this comment in the proto

```
 This field is expected to have many buckets,
    // especially zeros, so uint64 has been selected to ensure
    // varint encoding.
```

```
repeated fixed32 bucket_indexes;
repeated fixed64 bucket_counts;
```

No need to write out zeros

Reply:
@anuraaga @jsuereth @jkwatson I've made a few changes: 

I've removed the `List<Long>` and replaced it with these three methods with the aim of being flexible for implementations.
  - `long getBucketCountAt(int index)`
  - `int getNumBuckets()`
  - `int getStartIndex()`
 
@anuraaga so your for loop would look like 
```
for (int i = getStartIndex(); i < getNumBuckets(); i++)
```